### ISSUE TITLE
Improve clj-kondo to support reify / proxy / proxy+ 

### ISSUE THREAD

**Issue by ieugen** [clj-kondo/clj-kondo#1544]
Hello, 

To upvote this issue, give it a thumbs up. See [this list](https://github.com/clj-kondo/clj-kondo/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc) for the most upvoted issues.

[ To keep development of this project going, consider sponsoring. If you are
already a sponsor, thank you! You can remove this message. ]

**Is your feature request related to a problem? Please describe.**

Working with clj-kondo and proxy+ I found from @borkdude that clj-kondo does not fully support proxy / proxy+ macros and it fails to see unused symbols where there should be an error for unused symbols.

Bellow there is a simple sample code that fails to lint properly. 
`unused-symbol-not-detected` is not reported as it should.
```
(ns proxy-kondo)

;; clj-kondo --lint proxy-kondo.clj
;; proxy-kondo.clj:10:20: warning: unused binding a
;; linting took 20ms, errors: 0, warnings: 1

(let [p (proxy [java.io.InputStream] []
          (read
            ([] 
             (let [a (count unused-symbol-not-detected)]
               1))
            ([^bytes bytes] 2)
            ([^bytes bytes off len] 3)))]
  (println (.read p))
  (println (.read p (byte-array 3)))
  (println (.read p (byte-array 3) 0 3)))
```
For my use case I use proxy+ and I think this could also be supported with the right building blocks. 
https://github.com/redplanetlabs/proxy-plus/ 

`proxy+` seems to be very similar to many reify calls ?! 

**Describe the solution you'd like**

I don't know how the solution would look like but it should do proper linting for reify, proxy and maybe proxy+ :) . 

**Describe alternatives you've considered**
```
:lint-as {com.rpl.proxy-plus/proxy+ clojure.core/proxy}
```

**Additional context**

Sample code for proxy+ in project 
https://github.com/ieugen/ofbiz-tooling-ro/blob/61cf0b033e49d278fb841e86bee3cda694ae2e41/development/src/dev/schema.clj
